sched/semaphore/sem_holder.c:  Remove DEBUGPANIC

patacongo opened the pull request:
## Summary

There is a DEBUGPANIC in some logic.  This happens if a a task exists at certain points with priority inheritance enabled.  This event was not expected in the original design (although logic was provided to support it).  Since, apparently, it does happen, the DEBUGPANIC must be removed.

## Impact

Should eliminate a crash found by Brennan Ashton using the X86_64 Qemu port

## Testing

btashton commented:
This makes sense to me.

Should we also remove the first line of the assert in `nxsem_restorebaseprio` that was disabled and re-enable it. I think this explains the behavior.  I think the second half of the assert is correct.

patacongo commented:
> 
> 
> This makes sense to me.
> 
> Should we also remove the first line of the assert in `nxsem_restorebaseprio` that was disabled and re-enable it. I think this explains the behavior. I think the second half of the assert is correct.

I removed the code because I don't like commented out code.  It just leaves questions for the next person who deals with the code (and no one to answer the questions).  It is always better to be courageous and remove wrong code.

I did re-enable the debug output because I think it still has valid information.  I dropped from the error to a warning.  I think the output has value if you are debugging with scheduler debug turned off.. although perhaps.  It is not a error any more.  A warning is for unusual events that you might want to know about while debugging.  info() is just chit-char, flow of control stuff.

So generally, my preference would be to leave the warning in place because it provides information for people who are silly enough to use scheduler debug output.  I really doesn't work.  If you enable the scheduler debug output, the system becomes very unstable.  You can't really use the OS to monitor itself in the that way.  syslog output also requires the scheduler.

patacongo commented:
These are bogus build failures.  The macOS builds fail a large percentage of the time when there is no problem with the changes.  I have gotten so that I ignore these tests because the are not reliable indicators of anything.  I will restart the checks.  maybe we will be lucky next time.
